airflow/api_fastapi/execution_api/routes/xcoms.py

```diff
@@ -213,18 +198,6 @@ def set_xcom(
             },
         )
 
-    # TODO: This is in-efficient. We json.loads it here for BaseXCom.set to then json.dump it!
-    try:
-        json.loads(value)
-    except json.JSONDecodeError:
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail={
-                "reason": "invalid_format",
-                "message": "XCom value is not a valid JSON-formatted string",
-            },
-        )
-
```

Comment:
This was done to replicate what `JsonValue` does internally

https://docs.pydantic.dev/latest/api/types/#pydantic.types.JsonValue

Reply:
Yeah, not needed anymore now that the model in the response is JsonValue itself.

Reply:
Yep, remove the unnecessary :)